src/test/sh/Plugin.Testing.Stub-1.0.0/usr/opt/EXAplugins/Testing.Stub-1.0.0/exaoperation-gate/plugin-functions

```diff
@@ -1,32 +1,48 @@
-#!/usr/bin/env python3
-import sys, os, base64
-backup_home = "/opt/testing-stub"
-cfg_file = "%s/sync.cfg" % backup_home
-key_file = "%s/key" % backup_home
+#!/bin/bash
 
+readonly script_path="$(dirname $(readlink -f $0))"
 
-# definitions - interfaces
-plugin_functions = {
-    'INSTALL': 'Install plugin.',
-    'UNINSTALL': 'Install plugin.',
-    'STATUS': 'Show status of plugin (not installed, activated, not activated).',
-    'START': 'Start plugin service',
-    'STOP': 'Stop plugin service',
-    'RESTART': 'Restart the service',
-    'ON_BOOT': 'Undocumented'
+declare -A functions
+
+functions[INSTALL]='Install plugin.'
+functions[UNINSTALL]='Install plugin.'
+functions[STATUS]='Show status of plugin (not installed, activated, not activated).'
+functions[START]='Start plugin service'
+functions[STOP]='Stop plugin service'
+functions[RESTART]='Restart the service'
+functions[ON_BOOT]='Undocumented'
+
+usage() {
+    echo "Usage: $0 <function name> <parameter>"
+    echo "       $0 [--show-functions]"
 }
 
-# implement show-functions
-if len(sys.argv) == 2 and sys.argv[1] == '--show-functions':
-    print("\n".join(["%s:%s" % (X, Y) for X, Y in plugin_functions.items()]))
-    sys.exit(0)
-elif len(sys.argv) != 3:
-    print("Wrong call to plugin wrapper.")
-    sys.exit(1)
-elif sys.argv[1].upper() not in plugin_functions:
-    print("Unknown function '%s'" % sys.argv[1])
-    sys.exit(1)
+exit_abnormal() {
+    usage
+    exit 1
+}
+
+list_functions() {
+    for key in "${!functions[@]}"; do
+        echo " * $key: ${functions[$key]}"
```

Comment:
Please use same output format as done in the python version:
- It is possible that original EXAoperation actually parses the output to pre-validate incoming XML/rpc calls.
- We do not want potential plugin writers to pick up a non-working example

Reply:
Understood. Makes sense.

I wanted to improve usability, but you are right, people might rely on parsing the output.